src/opentelemetry.erl

```diff
@@ -106,12 +112,42 @@
 set_default_tracer(Tracer) ->
     persistent_term:put({?MODULE, default_tracer}, Tracer).
 
+set_default_context_manager(ContextModule) ->
+    persistent_term:put({?MODULE, context_manager}, ContextModule).
```

Comment:
would it make sense to guard that this is actually a module (or an atom if we don't check for loading)? Like there's a risk of setting a bad value and insta-killing all the things, no?

In fact looking at the call below, the expected form is  `{Module, Args}` and asserting there could prevent huge mistakes.

Reply:
Agh yea, very good catch.

Reply:
So it is actually used as just a module atom. Clearly need some typespecs so dialyzer will catch this shit where `get_*` is used.

I'm now unsure if there should be a tuple or not. Depends on if some state (that can't be updated because it is in a persistent term) is needed. I think I have an idea, will update this in a bit.